capstone/src/main/resources/static/script.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Retrieves events from server
+ */
+function getEvents() {
+  fetch('/list-events').then(response => response.json()).then((events) => {
```

Comment:
For these mappings maybe it is a good idea to use relative paths instead of absolute paths (remove the "/" in the front) but I'm not sure if spring-boot supports relative paths.

Reply:
When specified from Javascript, the relative vs absolute path will affect which back-end URL will be requested. Let's say, your domain is _collegeevents.com_, and your application sits in the very root of the domain. In this case, the distinction between an absolute and a relative URL won't matter -- either way, if you refer to "event/list" or "/event/list", the fetch function will contact _collegeevents.com/event/list_.

However, let's say you have multiple applications running on the same domain, and you want the event finding/rating application to live at _collegeevents.com/eventsapp_. In this case, "event/list" will go to _collegeevents.com/eventsapp/event/list_, but "/event/list" will go to the root of the domain -- _collegeevents.com/event/list_ where it won't find any of the controllers (since the server-side app listens to URLs under the `event` grouping.

For the purposes of the current app, you can do it either way, just be consistent. I wanted you to be aware of the distinction because you'll some day run into troubleshooting why your events go to the root of the domain.